File: src/main/java/nz/zephire/exceptions/exceptions/Exception.java
package nz.zephire.exceptions.exceptions;

import org.joda.time.LocalDate;
import org.joda.time.LocalTime;

public class Exception {

    private final LocalDate date;
    private final LocalTime start;
    private final LocalTime end;
    private final String approver;
    private final String reason;
    private final String description;

    public Exception(LocalDate date, LocalTime start, LocalTime end,
            String approver, String reason, String description) {

        this.date = date;
        this.start = start;
        this.end = end;
        this.approver = approver;
        this.reason = reason;
        this.description = description;
    }

    public LocalDate getDate() {
        return date;
    }

    public LocalTime getStart() {
        return start;
    }

    public LocalTime getEnd() {
        return end;
    }

    public String getApprover() {
        return approver;
    }

    public String getReason() {
        return reason;
    }

    public String getDescription() {
        return description;
    }
}

File: src/main/java/nz/zephire/exceptions/exceptions/Exception.kt
package nz.zephire.exceptions.exceptions

import org.joda.time.Duration
import org.joda.time.LocalTime
import org.joda.time.Period

open class Exception(val start: LocalTime, val end: LocalTime,
                     val approver: String, val reason: String, val description: String) {

    //    LocalTime a = new LocalTime(5,30);
    //    LocalTime b = new LocalTime(6, 0);
    //    Period p = new Period(a, b);
    //    p.toStandardDuration().getStandardMinutes(); // TODO: length of exception/differences

    val length: Duration = Period(start, end).toStandardDuration()
}
